applicationinsights-react-native/src/ReactNativeBasePlugin.ts

```diff
@@ -0,0 +1,316 @@
+                if (_device) {
+                    item.ext = item.ext || {};
+                    item.ext.device = item.ext.device || ({} as IDevice);
+                    if (isString(_device.id)) {
+                        item.ext.device.localId = _device.id;
+                    }
+                    if (isString(_device.model)) {
+                        item.ext.device.model = _device.model;
+                    }
+                    if (isString(_device.deviceClass)) {
+                        item.ext.device.deviceClass = _device.deviceClass;
+                    }
+                }
+            }
+
+            function _getGlobal(): any {
+                if (typeof global !== strShimUndefined && global) {
+                    return global as any;
+                }
+
+                return getGlobal() as any;
+            }
+
+            _self._setExceptionHandler = () => {
+                const _global = _getGlobal();
+                if (_global && _global.ErrorUtils) {
+                    // intercept react-native error handling
+                    _defaultHandler = (typeof _global.ErrorUtils.getGlobalHandler === "function" && _global.ErrorUtils.getGlobalHandler()) || _global.ErrorUtils._globalHandler;
+                    _global.ErrorUtils.setGlobalHandler(_trackException);
+                }
+            }
+
+            function _resetGlobalErrorHandler() {
+                const _global = _getGlobal();
+                if (_global && _global.ErrorUtils && _global.ErrorUtils.getGlobalHandler() === _trackException) {
+                    _global.ErrorUtils.setGlobalHandler(_defaultHandler || null);
+                }
+            }
+
+            // default global error handler syntax: handleError(e, isFatal)
+            function _trackException(e, isFatal) {
+                const exception: IExceptionTelemetry = { exception: e, severityLevel: eSeverityLevel.Error };
+
+                if (_analyticsPlugin) {
+                    _analyticsPlugin.trackException(exception);
+                } else {
+                    _throwInternal(_self.diagLog(),
+                        eLoggingSeverity.CRITICAL, _eInternalMessageId.TelemetryInitializerFailed, "Analytics plugin is not available, ReactNative plugin telemetry will not be sent: ");
+                }
+
+                // call the _defaultHandler - react native also gets the error
+                if (_defaultHandler) {
+                    _defaultHandler.call(global, e, isFatal);
+                }
+            }
+
+            // Test Hooks
+            (_self as any)._config = _config;
+            (_self as any)._getDbgPlgTargets = () => {
+                return [_device, _deviceInfoModule, _config];
+            }
+        });
+    }
+
+    protected getDeviceInfoModule(_deviceInfoModule): IDeviceInfoModule {
+        if (!_deviceInfoModule) {
+            console.log("ReactNativePlugin: Please provide device-info");
```

Comment:
We generally don't emit directly to the console, so we should use something like the following 

`_warnToConsole(_self.diagLog(), "Failed to get DeviceInfo: " + getExceptionName(e) + " - " + dumpObj(e));`

And because this will be automatically called as part of the collecting the device, it should also mention that they can disable this by the `disableDeviceCollection`